keps/sig-node/1664-node-ips/README.md

```diff
@@ -0,0 +1,767 @@
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+- [ ] Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] KEP approvers have approved the KEP status as `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+There are problems with the contents and interpretation of
+`Node.Status.Addresses` that currently break or complicate some IPv6
+and dual-stack scenarios.
+
+An earlier version of this KEP proposed deprecating
+`Node.Status.Addresses` in favor of a new field, but various issues
+make this complicated and to some extent pointless.
+
+This KEP now proposes deprecating kubelet's `--node-ip` argument and
+replacing it with a new slightly-different `--node-ips` argument which
+will result in `Node.Status.Addresses` reliably indicating IPv4 and
+IPv6 addresses according to the user's wishes.
+
+## Motivation
+
+### Goals
+
+1. Assign dual-stack `Pod.Status.PodIPs` to host-network Pods on nodes
+   that have both IPv4 and IPv6 IPs, so they can be targeted as
+   endpoints of IPv4, IPv6, or dual-stack Services. (This is
+   independent of the question of _how_ the node ends up with both
+   IPv4 and IPv6 IPs in `Node.Status.Addresses`.)
+
+2. Make the necessary changes to kubelet to allow bare-metal clusters
+   to have dual-stack node IPs (either auto-detected or specified on
+   the command line) rather than limiting them to a single node IP.
+
+3. Define how cloud providers should handle IPv4 and IPv6 node IPs in
+   different cluster types (single-stack IPv4, single-stack IPv6,
+   dual-stack) so as to enable IPv6/dual-stack functionality in
+   clusters that want it without accidentally breaking old IPv4-only
+   clusters. Update at least a few cloud providers to obey the new
+   rules. File issues against the remaining cloud providers pointing
+   out what needs to be done.
+
+4. Make built-in cloud providers and external cloud providers behave
```

Comment:
The problem is that they can only do that if the cloud wg in question is keeping the external cloud provider up to date with the built-in. At least for the case of Azure, this is not true.

But maybe? I guess this would be a decision for sig-cloud